src/node_messaging.cc

```diff
@@ -532,6 +553,11 @@ MessagePort* MessagePort::New(
     return nullptr;
   MessagePort* port = new MessagePort(env, context, instance);
   CHECK_NOT_NULL(port);
+  if (port->async_.data == nullptr) {
+    port->Close();
```

Comment:
Shouldn't this `close` happen in the MessagePort as a part of failure-handling when we failed to construct it?

IMO, not blocking but I think it would be better to have some sort of `init` function (or maybe override `operator new` if that's possible here?) that could return the status of the initialization than resort to checking properties like this.

Reply:
@lundibundi I head what you’re saying, but I’m not a fan of having half-initialized objects after the constructor has run…

I’ve updated the code to move the close call into the constructor, and made it private to make it clear that `MessagePort::New()` is the way to construct `MessagePort`s. Can you take another look and let me know what you think?

Reply:
Yeah, half initialized objects is a trade-off in this case.
> I’ve updated the code to move the close call into the constructor, and made it private to make it clear that MessagePort::New() is the way to construct MessagePorts. 

LGTM.